core/src/main/java/org/geysermc/geyser/entity/type/player/SessionPlayerEntity.java

```diff
@@ -226,7 +238,10 @@ public void setFlags(ByteEntityMetadata entityMetadata) {
     @Override
     protected void setSprinting(boolean value) {
         super.setSprinting(value);
-        session.setSprinting(value);
+
+        // Since we're using the sprinting value here to decide whether to send a sprinting packet
+        // Don't make it so that it can be changed by the flag, only by START_SPRINT and STOP_SPRINT that the player sent.
+        // session.setSprinting(value);
```

Comment:
uhm this is commented out? in which case this override could be removed entirely